### PR TITLE
Member entitlements

### PR DESCRIPTION
**Pull request by Macroz**
Implements granting entitlements for members after licenses are accepted and application approved from #653. Entitlements are also removed when a member is removed or the application is closed. The entitlement tests are moved into their own proper places.

### CODE REVIEW

File: src/clj/rems/application/model.clj [hunk 1]
```diff
@@ -116,12 +131,7 @@
 (defmethod calculate-permissions :application.event/approved
   [application _event]
   (-> application
-      (permissions/set-role-permissions (update closed-permissions
-                                                :handler set/union #{:application.command/add-member
-                                                                     :application.command/remove-member
-                                                                     :application.command/invite-member
-                                                                     :application.command/uninvite-member
-                                                                     :application.command/close}))))
+      (permissions/set-role-permissions approved-permissions)))
```
Comment (foxlynx) [inline src/clj/rems/application/model.clj]: Good that the approved permissions are now in separate definition. 👍

File: src/clj/rems/application/model.clj [hunk 2]
```diff
@@ -167,7 +178,7 @@
              :application/licenses (map (fn [license]
                                           {:license/id (:license/id license)})
                                         (:application/licenses event))
-             :application/accepted-licenses {"applicant" #{}}
+             :application/accepted-licenses {}
```
Comment (foxlynx) [inline src/clj/rems/application/model.clj]: Why did this change?

File: src/clj/rems/db/entitlements.clj [hunk 3]
```diff
@@ -62,30 +62,56 @@
           (log/warnf "Post failed: %s", response))
         (db/log-entitlement-post! {:target target :payload json-payload :status status})))))
 
+(defn- accepted-licenses? [application user-id]
+  (every? (or (get (:application/accepted-licenses application)
+                   user-id)
+              #{})
+          (map :license/id (:application/licenses application))))
```
Comment (foxlynx) [inline src/clj/rems/db/entitlements.clj]: Not sure if I understand this correctly, but isn't this true also for an empty hash-set?

File: test/clj/rems/api/test_entitlements.clj [hunk 4]
```diff
@@ -14,3 +15,39 @@
...
+                     handler
+                     read-body)]
+        (is (= 2 (count data)))))
+
+    (testing "just for alice"
+      (let [data (-> (request :get "/api/entitlements?user=alice")
+                     (authenticate api-key "developer")
+                     handler
+                     read-body)]
+        (is (= 1 (count data)))))
```
Comment (foxlynx) [inline test/clj/rems/api/test_entitlements.clj]: Maybe do some assert for the content of the data as well?

File: src/clj/rems/application/model.clj (hunk 2 above)
Comment (Macroz) [reply src/clj/rems/application/model.clj]: This was completely wrong. There should not be accepted licenses for the fixed value called "applicant" for every application. There should only be the correct ones that come later.

File: src/clj/rems/db/entitlements.clj (hunk 3 above)
Comment (Macroz) [reply src/clj/rems/db/entitlements.clj]: First part is geting the accepted application licenses for a particular user, or empty set if there is no user there can be no accepted licenses.

Then the second part is that we check that every license in the application is acceted. i.e. in the set of accepted application licenses of the user..
```
(every? #{} [1 2 3]) => false
```

File: test/clj/rems/api/test_entitlements.clj (hunk 4 above)
Comment (Macroz) [reply test/clj/rems/api/test_entitlements.clj]: This is an old test that I just moved here. There is not really a good spec on what should be returned so anything goes here. The more interesting tests about content are in the `db/test_applications.clj` so far

**Pull request merged**